### DISCUSSION TITLE
NotifyIcon without a MainWindow

### DISCUSSION THREAD

**Discussion by JustArion** [lepoco/wpfui#719] [Q&A]
Is it possible to generate a NotifyIcon without the MainWindow being shown in a noticeable way. I looked through some of the code and it seems to rely on hooking into a window's message procs. Is there a recommended approach to having a NotifyIcon without a visible window tied to it.

**Comment by JustArion** [comment]
I have tried creating a Window with `WindowState="Minimized"` but the obvious issue with it is there's an icon in the taskbar.
If I add `ShowInTaskbar="False"` it shows a partial window at the bottom left.

<img src="https://cdn.discordapp.com/attachments/883435300880261120/1136689754914701404/6007a3de-a15b-4a03-9b3f-3d7a99bcd3e6_03-08-2023.png"/>

**Comment by TheJoeFin** [comment]
One solution I am trying is to make a window which holds the NotifyIcon. To hide the window there are a few tricks I'm doing:
- WindowStyle="ToolWindow"      (keeps the window out of Alt+Tab)
- OnActivated(...) { Hide(); }            (this will always hide the window any time it is shown like when the icon is clicked)
- ShowInTaskbar="False"               (you knew this one already)

Right now I'm experimenting with this and will let you know how it goes.
